include/oneapi/dpl/pstl/utils.h

```diff
@@ -695,6 +695,21 @@ __iterators_possibly_equal(_Iterator1 __it1, _Iterator2 __it2)
     }
 }
 
+// Conditionally sets type to _IsSpirvT if oneDPL is being compiled to a SPIR-V target with the SYCL backend and the negation otherwise.
+template <typename _IsSpirvT>
+struct __spirv_target_selector :
+#if _ONEDPL_DETECT_SPIRV_COMPILATION
+    _IsSpirvT
+#else
+    ::std::bool_constant<!_IsSpirvT::value>
+#endif
+{
+};
```

Comment:
I am not sure I agree with the change to just one parameter. You have made an assumption that the parameter is always a bool_constant, and that the values are always opposite. That's sufficient for the current use case, but loses some generality and orthogonality of the original design (which could have been applied to any type, e.g. float and double). Also the code readability is worse, because `using _Commutative = oneapi::dpl::__internal::__spirv_target_conditional<::std::false_type, ::std::true_type>` is more self-explaining than `using _Commutative = oneapi::dpl::__internal::__spirv_target_selector<::std::false_type>`.

Reply:
I agree that `using _Commutative = oneapi::dpl::__internal::__spirv_target_conditional<::std::false_type, ::std::true_type>` is more readable in the code that uses this utility and there seems to be stronger consensus to implement this in a more generic way to enable non-bool types.

@rarutyun Do you have any objection to going back to the original design of the `__spirv_target_selector` struct? This would add back the second template parameter type and define a `type` alias in the body in of the struct.

Reply:
I had an offline discussion with @rarutyun and have reverted back to using the previous implementation of `__spirv_target_selector`.